Sources/Purchasing/StoreKit1/ProductsFetcherSK1.swift

```diff
@@ -188,6 +188,12 @@ extension ProductsFetcherSK1: SKProductsRequestDelegate {
         }
     }
 
+    func clearCache() {
+        self.queue.async {
```

Comment:
I feel like using dispatch barrier might be a good move here, so all previous requests can complete before we clear. 
`.async(flags: .barrier)`

Reply:
Mmm...it's a [serial queue](https://github.com/RevenueCat/purchases-ios/blob/d302f8ffd66cc3d5feeed9bc694e5a90b80708bb/Sources/Purchasing/StoreKit1/ProductsFetcherSK1.swift#L27) and as far as I know the barrier flag only affects to concurrent queues, doesn't it? Let me know if I'm wrong.

Reply:
No no, you're right 😄, I'll wait for Nacho to do one more pass, but this looks great.

Reply:
Ah ok ok 🙂 thanks for the review 🤙🏼